bin/export-record-barcodes

```diff
@@ -35,39 +54,142 @@ class HuskyProject(redcap.Project):
         super().__init__(REDCAP_API_URL, REDCAP_API_TOKEN_UW_REOPENING, project_id)
         self.lang = 'en'
         self.purview = 'irb'
+        # There are currently two events for this project: Enrollment and
+        # Encounter events, both part of arm 1. None of the Enrollment event
+        # isntruments have any barcode fields, so include only the Encounter
+        # event_id.
+        #
+        # -kfay, 30 Sep 2020
+        self.event_id_map = {'encounter_arm_1': 742155}
 
 
 class ScanProject(redcap.Project):
     lang: str
     purview: Optional[str]
 
-    def __init__(self, project_id: str, lang: str, purview: str = None) -> None:
+    def __init__(self,
+        project_id: int,
+        lang: str,
+        purview: str,
+        event_id_map: Dict[int, int]) -> None:
+        """
+        Creates a SCAN Project object given the REDCap project PID *project_id*,
+        an ISO language code *lang*, a study *purview* (either IRB or Public
+        Health), and a REDCap project *event_id_map* that contains a map of a
+        project's unique event name to the associated `event_id` in REDCap.
+        """
         super().__init__(REDCAP_API_URL, REDCAP_API_TOKEN_SCAN(lang, purview), project_id)
         self.lang = lang
         self.purview = purview
-
+        self.event_id_map = event_id_map
 
 def main():
+    # Unfortunately, the REDCap API does not expose the event_id attribute which
+    # is required to generate deep links into a specific "page" in a REDCap
+    # project. The event_id changes per project event, so hard code the
+    # relationship between unique event name and event_id for each REDCap
+    # project.
     projects = [
         # SCAN (public health action)
-        ScanProject(20759, "en", "ph"),
-        ScanProject(21520, "es", "ph"),
-        ScanProject(21521, "zh-Hant", "ph"),
+        ScanProject(20759, "en", "ph", {
+            'priority_arm_1': 732107,
+            'symptomatic_arm_2': 732126,
+            'asymptomatic_arm_3': 732127,
+        }),
+        ScanProject(21520, "es", "ph", {
+            'priority_arm_1': 735113,
+            'symptomatic_arm_2': 735114,
+            'asymptomatic_arm_3': 735115,
+        }),
+        ScanProject(21521, "zh-Hant", "ph", {
+            'priority_arm_1': 735116,
+            'symptomatic_arm_2': 735117,
+            'asymptomatic_arm_3': 735118,
+        }),
```

Comment:
Non-blocking, but with the addition of all of these event maps, it'd be even nicer to extract out this data into a separate NDJSON/YAML file.

Reply:
I agree!

Reply:
I wrote a comment about this in a new commit.